wfl/Database/IDBManager.cs

```diff
@@ -1,16 +1,15 @@
 ﻿using System.Collections.Generic;
-using Symphono.Wfl.Models;
 using System.Threading.Tasks;
+using Symphono.Wfl.Models;
+using System;
 
 namespace Symphono.Wfl.Database
 {
     public interface IDBManager
     {
-        Task<Restaurant> InsertRestaurantAsync(RestaurantDto r);
-        Task<IEnumerable<Restaurant>> GetAllRestaurantsAsync();
-        Task<Restaurant> GetRestaurantWithIdAsync(string Id);
-        Task<Restaurant> UpdateRestaurantAsync(string id, RestaurantDto restaurant);
-        Task<IEnumerable<FoodOrder>> GetAllFoodOrdersAsync();
-        Task<FoodOrder> InsertFoodOrderAsync(FoodOrderDto order);
+        Task<T> InsertEntityAsync<T>(T entity) where T : IEntity, new();
+        Task<IEnumerable<T>> GetAllEntitiesAsync<T>() where T : IEntity, new();
+        Task<T> GetEntityWithIdAsync<T>(string id) where T : IEntity, new();
```

Comment:
Rename this method to `GetEntityByIdAsync()`. `Withid` could imply the id can be nullable.

[ ] Rename method